content/6_Advanced/Wavelet.mdx

```diff
@@ -32,9 +30,232 @@ Like a segment tree on values rather than indices.
 	</Resource>
 </Resources>
 
-### Solution - Range K-th Smallest
+Suppose you want to support the following queries:
+
+- Given a range $l$, $r$ count the number of occurrences of value $x$.
+- Given a range $l$, $r$ find the $k$ smallest element
+
+With wavelet tree, you can easily support those queries in $log(M)$ time.
+when M is the maximum value in the array.
+
+## Wavelet tree structure
+
+A wavelet tree is a binary tree where each node represents a range of values.
+The root node covers the entire range, and each subsequent level splits the
+range into two halves.
+
+We are going to maintain a segment tree over values instead of indices. Each segment will contain indices whose values lie within the segment's range. We'll save those indices in a vector. Notice that an index can be in at most $log(M)$ segments
+
+<Spoiler title="Wavelet tree Visualization">
+
+Let's say our array is: $[3,5,3,1,2,2,3,4,5,5]$
```

Comment:
why does your visualization not contain the same array as the example you give?

Reply:
Oops, this is my bad. I misread the second level of nodes.

Regarding the visualization, this is very minor but $7$ should be in brackets instead of parentheses, just to be consistent.